ykrt/src/mt.rs

```diff
@@ -1279,6 +1283,30 @@ enum TransitionControlPoint {
     },
 }
 
+/// Why did we abort tracing?
+#[derive(Debug)]
+enum AbortKind {
+    /// While tracing we fell back from an interpreter to a JIT frame.
+    BackIntoExecution,
+    /// While tracing, we encountered a compiled trace.
+    EncounteredCompiledTrace,
+    /// Tracing continued while the interpreter frame address changed.
+    OutOfFrame,
+    /// We unrolled a loop (i.e. we traced a [Location] more than once).
```

Comment:
I had to think about this one a bit.

 - If we encounter the same location twice and it's the place we started tracing we stop tracing and compile.
 - If we enounter the same location twice and it's *not* the place we started tracing, this is indicative of an inner loop being unrolled, which is what the `Unrolled` abort kind is for.
 
 Is that right?

Reply:
You're right.

[Just to be clear: this PR does not change behaviour, it just makes it more obvious what happened.]